x-pack/plugins/monitoring/server/lib/__tests__/helpers.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const response = {
+  hits: {
+    hits: [
+      {
+        _source: {
+          beats_stats: {
+            timestamp: 1515541592880,
+            beat: {
+              uuid: 'fooUuid',
+              host: 'beat-listing.test',
+              name: 'beat-listing.test-0101',
+              type: 'filebeat',
+              version: '6.2.0'
+            },
+            metrics: {
+              beat: {
+                memstats: {
+                  memory_alloc: 2340
+                }
+              },
+              libbeat: {
+                output: {
+                  type: 'Redis',
+                  write: {
+                    bytes: 140000,
+                    errors: 8,
+                  },
+                  read: {
+                    errors: 3,
+                  }
+                },
+                pipeline: {
+                  events: {
+                    total: 23000
+                  }
+                }
+              }
+            }
+          }
+        },
+        inner_hits: {
+          earliest: {
+            hits: {
+              hits: [
+                {
+                  _source: {
+                    beats_stats: {
+                      timestamp: 1515534342000,
+                      metrics: {
+                        libbeat: {
+                          output: {
+                            write: {
+                              bytes: 4000,
+                              errors: 3,
+                            },
+                            read: {
+                              errors: 1,
+                            }
+                          },
+                          pipeline: {
+                            events: {
+                              total: 2300
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        }
+      }
+    ]
+  }
+};
+
+export const defaultResponseSort = (handleResponse) => {
+  const responseMulti = { hits: { hits: [] } };
+  const hit = response.hits.hits[0];
+  const version = ['6.6.2', '7.0.0-rc1', '6.7.1'];
+
+  for (let i = 0, l = version.length; i < l; ++i) {
+    const newBeat = JSON.parse(JSON.stringify({ ...hit }));
```

Comment:
Isn't this effectively the same as: `const newBeat = { ...hit };` or `const newBeat = hit;`?

Reply:
I tried it with the spread/assign, but it failed, because some objects in the base object are still a reference and not a copy. This was the only way to ensure un-referenced copy of the object and all its children

Reply:
Ok, could we add a comment to that effect? It looks like this method was employed elsewhere in Kibana with an accompanying comment. Thanks for explaining!